From 601c151939427e3dfa43ad87217de7062676c495 Mon Sep 17 00:00:00 2001
From: Stoica Cosmin-Stefan <cosmin.stoica@nxp.com>
Date: Fri, 4 Feb 2022 16:49:40 +0000
Subject: [PATCH 27/64] imx8mp: freertos: Add ENET_QOS to the industrial cell
 config

Signed-off-by: Stoica Cosmin-Stefan <cosmin.stoica@nxp.com>
[Adapt to cell config helper format, fix some attributes]
Signed-off-by: Stephane Viau <stephane.viau@nxp.com>
---
 configs/arm64/imx8mp-harpoon-freertos-industrial.c | 6 +++++-
 configs/arm64/imx8mp-harpoon-rtos.h                | 2 ++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/configs/arm64/imx8mp-harpoon-freertos-industrial.c b/configs/arm64/imx8mp-harpoon-freertos-industrial.c
index 338bfe82..fc94acbe 100644
--- a/configs/arm64/imx8mp-harpoon-freertos-industrial.c
+++ b/configs/arm64/imx8mp-harpoon-freertos-industrial.c
@@ -25,11 +25,13 @@
 	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */  \
 	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */   \
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
+	MMIO_REGION_RWS(0x30230000, 0x30230000, KB(64)),   /* GPIO4 */  \
 	MMIO_REGION_RWS(0x30240000, 0x30240000, KB(64)),   /* GPIO5 */  \
 	MMIO_REGION_RWS(0x30340000, 0x30340000, KB(64)),   /* GPR */    \
 	MMIO_REGION_RWS(0x30360000, 0x30360000, KB(4)),    /* ANA_PLL */\
 	MMIO_REGION_RWS(0x30380000, 0x30380000, KB(64)),   /* CCM */    \
-	MMIO_REGION_RW( 0x308c0000, 0x308c0000, KB(64))    /* FLEXCAN1*/
+	MMIO_REGION_RW( 0x308c0000, 0x308c0000, KB(64)),   /* FLEXCAN1*/\
+	MMIO_REGION_RW( 0x30bf0000, 0x30bf0000, KB(64))    /* ENET_Q */
 
 /* GIC */
 #define CONFIG_INMATE_IRQCHIPS_ADDR	(0x38800000)
@@ -45,6 +47,8 @@
 		RTOS_COMMON_IRQCHIPS_BITMAP4),             \
 	IRQCHIP1(CONFIG_INMATE_IRQCHIPS_ADDR,              \
 		/* interrupts 160..191 */                  \
+		IRQnMAP(ENET_QOS_PMT_IRQn) |               \
+		IRQnMAP(ENET_QOS_IRQn) |                   \
 		IRQnMAP(CAN_FD1_IRQn) | /* FLEXCAN1 */     \
 		IRQnMAP(CAN_FD1_ERROR_IRQn),               \
 		/* interrupts 192..223 */                  \
diff --git a/configs/arm64/imx8mp-harpoon-rtos.h b/configs/arm64/imx8mp-harpoon-rtos.h
index 6091484b..47cb99f1 100644
--- a/configs/arm64/imx8mp-harpoon-rtos.h
+++ b/configs/arm64/imx8mp-harpoon-rtos.h
@@ -20,6 +20,8 @@ typedef enum IRQn {
   GPT2_IRQn                    = 86,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   GPT1_IRQn                    = 87,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   I2S56_IRQn                   = 122,              /**< SAI5/6 Receive / Transmit Interrupt */
+  ENET_QOS_PMT_IRQn            = 166,              /**< ENET QOS TSN Interrupt from PMT */
+  ENET_QOS_IRQn                = 167,              /**< ENET QOS TSN LPI RX exit/Host System/RX/TX Channels[4:0] Interrupt */
   CAN_FD1_IRQn                 = 174,              /**< CAN-FD1 Interrupt from bus off/line error/RX warning/TX warning/wakeup/match in PN/timeout in PN/busoff done/FD error */
   CAN_FD1_ERROR_IRQn           = 175,              /**< CAN-FD1 Interrupt from correctable error/non correctable error int host/ non correctable error int internal */
 } IRQn_Type;
-- 
2.34.1

